src/gui/attributetable/qgsattributetablefiltermodel.cpp

```diff
@@ -226,11 +226,18 @@ void QgsAttributeTableFilterModel::setSelectedOnTop( bool selectedOnTop )
   if ( mSelectedOnTop != selectedOnTop )
   {
     mSelectedOnTop = selectedOnTop;
+    int column = sortColumn();
+    Qt::SortOrder order = sortOrder();
 
-    if ( sortColumn() == -1 )
-    {
-      sort( 0 );
-    }
+    // set default sort values if they are not correctly set
```

Comment:
Is this the right place to deal with this?
If this is sometimes setup the wrong way, it's probably not only sortOnTop that suffers from this?